fsw/cfe-core/src/sb/cfe_sb_api.c

```diff
@@ -1047,10 +1046,13 @@ int32 CFE_SB_UnsubscribeFull(CFE_SB_MsgId_t MsgId,CFE_SB_PipeId_t PipeId,
 
     /* if there are no subscriptions for this message id... */
     if(!CFE_SB_IsValidRouteIdx(RouteIdx)){
+        char    PipeName[OS_MAX_API_NAME] = {'\0'};
+
+        CFE_SB_GetPipeName(PipeId, PipeName);
         CFE_SB_UnlockSharedData(__func__,__LINE__);
```

Comment:
In this case the CFE_SB_GetPipeName is called while the SB global lock is still held.  This should probably be switched.  In all other cases it looks like CFE_SB_GetPipeName is invoked outside the lock, which is probably preferable, as it avoids a double lock with OSAL.  OSAL already has protections in case a delete occurs in close proximity to this call.

Reply:
Fixed!